phoenix-core/src/main/java/org/apache/phoenix/schema/PTable.java

```diff
@@ -728,6 +729,13 @@ private static int getReservedQualifier(byte[] bytes, int offset, int length) {
      * (use @getPhysicalTableName for this case) 
      */
     PName getParentTableName();
+
+    /**
+     * @return the logical name of the parent. In case of the view index, it is the _IDX_+logical name of base table
+     * Ex: For hierarchical views like tableLogicalName --> view1 --> view2, for view2, returns _IDX_+tableLogicalName
+     */
+    PName getParentLogicalName();
```

Comment:
Likewise with getParentName vs getParentLogicalName. If I'm reading the current logic right, getParentName _also_ returns only logical names. They're almost-but-not-quite the same thing. Can we consolidate? Otherwise we're going to create lots of subtle, really-hard-to-spot bugs going forward when people use the wrong one.